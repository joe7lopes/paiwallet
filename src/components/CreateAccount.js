import React from 'react';

class CreateAccount extends React.Component {

    render() {
        return (
            <section className="hero is-light">
                <div className="hero-body">

                    <h1 className="title">
                        Create Account
                    </h1>
                    <div className="field is-grouped">
                        <div className="control">
                            <input className="input" type="text" placeholder="First name"/>
                        </div>
                        <div className="control">
                            <input className="input" type="text" placeholder="Surname"/>
                        </div>
                    </div>
                    <div className="field">
                        <div className="control">
                            <input className="input" type="email" placeholder="Email"/>
                        </div>
                    </div>
                    <div className="field">
                        <div className="control">
                            <input className="input" type="password" placeholder="New Password"/>
                        </div>
                    </div>
                    <a className="button is-success is-medium">Success</a>
                </div>
            </section>
        );
    }
}

export default CreateAccount;