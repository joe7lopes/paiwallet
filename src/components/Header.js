import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import { startLogout } from '../actions/auth_action';

export const Header = ({startLogout}) => {
    return (
            <div className="nav-wrapper">
                <nav role="navigation">
                    <div className="nav-wrapper container">
                        <a className="brand-logo">PaiWallet</a>
                        <ul id="nav-mobile" className="right hide-on-med-and-down">
                            <li>
                                <Link to="/">Home</Link>
                            </li>
                            <li>
                                <Link to="/subscription">Subscription</Link>
                            </li>
                            <li>
                                <Link to="/topstocks">Top Stocks</Link>
                            </li>
                            <li>
                                <Link to="/admin">Admin</Link>
                            </li>
                            <li>
                                <Link to="/disclaimer">Disclaimer</Link>
                            </li>
                            <li>
                                <a onClick={startLogout}>Logout</a>
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>
    );
}


const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined,mapDispatchToProps) (Header);