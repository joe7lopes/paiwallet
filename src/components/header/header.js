import React from 'react';
import {Link, NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import { startLogout } from '../../actions/users_action';
import { Dropdown, NavItem, SideNav, SideNavItem, Button } from 'react-materialize';


const menuItems = [
    {title: "Home", link:'/'},
    {title: "Subscription", link:'/subscription'},
    {title: "Top Stocks", link:'/topstocks'},
    {title: "Disclaimer", link:'/disclaimer'}
]

class Header extends React.Component{

    renderLinks = () => {
        return menuItems.map(item =>{
            return (
                <li key={item.title}>
                    <NavLink to={item.link} activeClassName="active" >{item.title}</NavLink>
                </li>
            );
        });
    }

    renderSideBarLinks = () => {
        return menuItems.map(item =>{
            return (
                    <Link key={item.title} to={item.link} style={{padding: "0 32px"}}>{item.title}</Link>
            );
        })
    }

    renderAdminLinks = () =>(
        <li>
            <NavLink to="/admin">Admin</NavLink>
        </li>
    );
    
    renderAccount = () => (
        <li>
           <Dropdown trigger={
                <a>Account<i className="material-icons right">arrow_drop_down</i></a>
            }>
                <NavLink to='/account'>My Profile</NavLink>
                <NavItem divider />
                <a onClick={this.props.startLogout}>Logout</a>
            </Dropdown>
        </li>
    );
    
    render(){
        const { isAuthenticated, isAdmin } = this.props;
        return (
                <div className="nav-wrapper">
                    <nav role="navigation">
                        <div className="nav-wrapper container">
                            <a className="brand-logo">PaiWallet</a>
                            <ul id="nav-mobile" className="right hide-on-med-and-down">
                                { this.renderLinks() }
                                { isAdmin && this.renderAdminLinks() }
                                { isAuthenticated && this.renderAccount() }
                            </ul>
                                <SideNav
                                    trigger={<a href="" className="button-collapse"><i className="material-icons">menu</i></a>}
                                    options={{ closeOnClick: true }}
                                    >
                                    { this.renderSideBarLinks() }
                                    <SideNavItem divider />
                                    <SideNavItem subheader>Account</SideNavItem>
                                    <Link style={{padding: "0 32px"}} to='/account'>My Profile</Link>
                                    <a style={{padding: "0 32px"}} onClick={this.props.startLogout}>Logout</a>
                                    
                                </SideNav>
                        </div>
                    </nav>
                </div>
        );
    }

}

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => ({
    isAuthenticated: !!state.users.loggedInUser,
    isAdmin: !!state.users.loggedInUser && !!state.users.loggedInUser.admin
});

export default connect(mapStateToProps, mapDispatchToProps) (Header);