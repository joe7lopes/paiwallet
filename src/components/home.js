import React from 'react';
import Footer from './footer';

class Home extends React.Component {

    render() {
        return (
            <div>
                <section>
                    <div className="home-container">
                        <div className="row container">
                            <div className="col s6">
                                <h1> PaiWallet</h1>
                                <h3>Fundos de investimento</h3>
                                <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Facilis, voluptatibus obcaecati. Ipsam rem commodi reiciendis unde cupiditate, iure tempora consequuntur tempore quas ut aperiam nulla quod adipisci provident eligendi qui.</p>
                                <div className="col s6">
                                    <a class="waves-effect waves-light btn-large"><i class="material-icons left">system_update</i>Apps Store</a>
                                </div>
                                <div className="col s6">
                                    <a class="waves-effect waves-light btn-large"><i class="material-icons left">adb</i>Play Store</a>
                                </div>
                            </div>
                            <div className="col s6 center">
                                signup
                            </div>
                        </div>
                    </div>
                </section>
                <Footer/>
            </div>
        );
    }
}

export default Home;