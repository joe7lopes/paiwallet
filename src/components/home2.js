import React from 'react';
import TopStocksPerformance from './topStocksPerformance';

class Home extends React.Component {

    render() {
        return (
            <div>
               <TopStocksPerformance/>
                <section className="hero is-primary">
                    <div className="hero-body">
                        <div className="container">
                            <nav className="level">
                                <div className="level-item has-text-centered">
                                    <div>
                                        <p className="heading">Tweets</p>
                                        <p className="title">3,456</p>
                                    </div>
                                </div>
                                <div className="level-item has-text-centered">
                                    <div>
                                        <p className="heading">Following</p>
                                        <p className="title">123</p>
                                    </div>
                                </div>
                                <div className="level-item has-text-centered">
                                    <div>
                                        <p className="heading">Followers</p>
                                        <p className="title">456K</p>
                                    </div>
                                </div>
                                <div className="level-item has-text-centered">
                                    <div>
                                        <p className="heading">Likes</p>
                                        <p className="title">789</p>
                                    </div>
                                </div>
                            </nav>
                        </div>
                    </div>
                </section>
            </div>
        );
    }
}

export default Home;