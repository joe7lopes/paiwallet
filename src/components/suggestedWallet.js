import React from 'react';
import Wallet from './wallet';

class SuggestedWallet extends React.Component {

    render() {
        return (
            <section>
                <div className="columns">
                    <div className="column">
                        <div className="content">
                        <Wallet/>
                        </div>
                    </div>
                    <div className="column">
                        <div className="content">
                            <h4 className="subtitle is-4">Your Bugget</h4>
                            <progress className="progress is-success" value="60" max="100">60%</progress>
                        </div>
                        <div className="content">
                            <h4 className="subtitle is-4">Risk Tolerance</h4>
                            <progress className="progress is-warning" value="60" max="100">60%</progress>
                        </div>
                    </div>
                </div>
            </section>
        );
    }
}

export default SuggestedWallet;