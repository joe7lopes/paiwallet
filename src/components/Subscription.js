import React from 'react';
import CreateAccount from './CreateAccount';
class Subscription extends React.Component {

    render() {
        return (
            <section className="section">
                <div className="columns">
                    <div className="column is-half">
                        some tiles
                    </div>
                    <div className="column is-half">
                        <CreateAccount/>
                    </div>
                </div>  
            </section>
        );
    }
}

export default Subscription;