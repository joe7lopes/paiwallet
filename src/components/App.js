import React from 'react';
import Header from './Header';
import Login from './login/Login';

class App extends React.Component{

    render(){
        return(
            <h1>HOME</h1>
        );
    }
}

export default App;