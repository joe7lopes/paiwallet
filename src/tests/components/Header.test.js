import React from 'react';
import { shallow } from '../config';
import { Header } from '../../components/header';

describe('<Header />', ()=> {

    // it('should render Header', ()=>{
    //     const wrapper = shallow(<Header />);
    //     expect(wrapper).toMatchSnapshot();
    // });

    // it('should call startLogout on button click',()=>{
    //     const startLogout = jest.fn();
    //     const wrapper = shallow(<Header startLogout={startLogout}/>);
    //     wrapper.find('button').at(1).simulate('click');
    //     expect(startLogout).toHaveBeenCalled();
    // });

});
