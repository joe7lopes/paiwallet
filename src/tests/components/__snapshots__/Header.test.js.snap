// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        role="navigation"
      >
        <div
          className="nav-wrapper container"
        >
          <a
            className="brand-logo"
          >
            PaiWallet
          </a>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/subscription"
              >
                Subscription
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/topstocks"
              >
                Top Stocks
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/admin"
              >
                Admin
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/disclaimer"
              >
                Disclaimer
              </Link>
            </li>
            <li>
              <button
                onClick={undefined}
              >
                Logout
              </button>
            </li>
          </ul>
        </div>
      </nav>,
      "className": "nav-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper container"
        >
          <a
            className="brand-logo"
          >
            PaiWallet
          </a>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/subscription"
              >
                Subscription
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/topstocks"
              >
                Top Stocks
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/admin"
              >
                Admin
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/disclaimer"
              >
                Disclaimer
              </Link>
            </li>
            <li>
              <button
                onClick={undefined}
              >
                Logout
              </button>
            </li>
          </ul>
        </div>,
        "role": "navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="brand-logo"
            >
              PaiWallet
            </a>,
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/subscription"
                >
                  Subscription
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/topstocks"
                >
                  Top Stocks
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/admin"
                >
                  Admin
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/disclaimer"
                >
                  Disclaimer
                </Link>
              </li>
              <li>
                <button
                  onClick={undefined}
                >
                  Logout
                </button>
              </li>
            </ul>,
          ],
          "className": "nav-wrapper container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PaiWallet",
              "className": "brand-logo",
            },
            "ref": null,
            "rendered": "PaiWallet",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/subscription"
                  >
                    Subscription
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/topstocks"
                  >
                    Top Stocks
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/admin"
                  >
                    Admin
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/disclaimer"
                  >
                    Disclaimer
                  </Link>
                </li>,
                <li>
                  <button
                    onClick={undefined}
                  >
                    Logout
                  </button>
                </li>,
              ],
              "className": "right hide-on-med-and-down",
              "id": "nav-mobile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/subscription"
                  >
                    Subscription
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Subscription",
                    "replace": false,
                    "to": "/subscription",
                  },
                  "ref": null,
                  "rendered": "Subscription",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/topstocks"
                  >
                    Top Stocks
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Top Stocks",
                    "replace": false,
                    "to": "/topstocks",
                  },
                  "ref": null,
                  "rendered": "Top Stocks",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/admin"
                  >
                    Admin
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Admin",
                    "replace": false,
                    "to": "/admin",
                  },
                  "ref": null,
                  "rendered": "Admin",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/disclaimer"
                  >
                    Disclaimer
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Disclaimer",
                    "replace": false,
                    "to": "/disclaimer",
                  },
                  "ref": null,
                  "rendered": "Disclaimer",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    onClick={undefined}
                  >
                    Logout
                  </button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Logout",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": "button",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          role="navigation"
        >
          <div
            className="nav-wrapper container"
          >
            <a
              className="brand-logo"
            >
              PaiWallet
            </a>
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/subscription"
                >
                  Subscription
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/topstocks"
                >
                  Top Stocks
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/admin"
                >
                  Admin
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/disclaimer"
                >
                  Disclaimer
                </Link>
              </li>
              <li>
                <button
                  onClick={undefined}
                >
                  Logout
                </button>
              </li>
            </ul>
          </div>
        </nav>,
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nav-wrapper container"
          >
            <a
              className="brand-logo"
            >
              PaiWallet
            </a>
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/subscription"
                >
                  Subscription
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/topstocks"
                >
                  Top Stocks
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/admin"
                >
                  Admin
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/disclaimer"
                >
                  Disclaimer
                </Link>
              </li>
              <li>
                <button
                  onClick={undefined}
                >
                  Logout
                </button>
              </li>
            </ul>
          </div>,
          "role": "navigation",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="brand-logo"
              >
                PaiWallet
              </a>,
              <ul
                className="right hide-on-med-and-down"
                id="nav-mobile"
              >
                <li>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/subscription"
                  >
                    Subscription
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/topstocks"
                  >
                    Top Stocks
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/admin"
                  >
                    Admin
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/disclaimer"
                  >
                    Disclaimer
                  </Link>
                </li>
                <li>
                  <button
                    onClick={undefined}
                  >
                    Logout
                  </button>
                </li>
              </ul>,
            ],
            "className": "nav-wrapper container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PaiWallet",
                "className": "brand-logo",
              },
              "ref": null,
              "rendered": "PaiWallet",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/subscription"
                    >
                      Subscription
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/topstocks"
                    >
                      Top Stocks
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/admin"
                    >
                      Admin
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/disclaimer"
                    >
                      Disclaimer
                    </Link>
                  </li>,
                  <li>
                    <button
                      onClick={undefined}
                    >
                      Logout
                    </button>
                  </li>,
                ],
                "className": "right hide-on-med-and-down",
                "id": "nav-mobile",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/subscription"
                    >
                      Subscription
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Subscription",
                      "replace": false,
                      "to": "/subscription",
                    },
                    "ref": null,
                    "rendered": "Subscription",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/topstocks"
                    >
                      Top Stocks
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Top Stocks",
                      "replace": false,
                      "to": "/topstocks",
                    },
                    "ref": null,
                    "rendered": "Top Stocks",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/admin"
                    >
                      Admin
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Admin",
                      "replace": false,
                      "to": "/admin",
                    },
                    "ref": null,
                    "rendered": "Admin",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/disclaimer"
                    >
                      Disclaimer
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Disclaimer",
                      "replace": false,
                      "to": "/disclaimer",
                    },
                    "ref": null,
                    "rendered": "Disclaimer",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      onClick={undefined}
                    >
                      Logout
                    </button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Logout",
                      "onClick": undefined,
                    },
                    "ref": null,
                    "rendered": "Logout",
                    "type": "button",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
